{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ed12860a-0395-4b5e-a38f-39b101c08102",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/tracker_store.py:1044: MovedIn20Warning: \u001b[31mDeprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. \u001b[32mTo prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to \"sqlalchemy<2.0\". \u001b[36mSet environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message.\u001b[0m (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)\n",
      "  Base: DeclarativeMeta = declarative_base()\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/validation.py:134: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html\n",
      "  import pkg_resources\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/pkg_resources/__init__.py:3117: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.\n",
      "Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages\n",
      "  declare_namespace(pkg)\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/pkg_resources/__init__.py:3117: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('ruamel')`.\n",
      "Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages\n",
      "  declare_namespace(pkg)\n",
      "2025-02-21 19:06:04 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.cli.train\u001b[0m  - Started validating domain and training data...\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/tensorflow/lite/python/util.py:52: DeprecationWarning: jax.xla_computation is deprecated. Please use the AOT APIs.\n",
      "  from jax import xla_computation as _xla_computation\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Loading domain from 'domain.yml' failed. Using empty domain. Error: 'Failed to load domain specification from '/home/moringa/capstone_project_group2/domain.yml'. File not found!'\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/.ipynb_checkpoints/rules-checkpoint.yml': \n",
      "Found intent 'goodbye' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/.ipynb_checkpoints/rules-checkpoint.yml': \n",
      "Found intent 'bot_challenge' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/rules.yml': \n",
      "Found intent 'goodbye' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/rules.yml': \n",
      "Found intent 'bot_challenge' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "2025-02-21 19:06:07 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating intents...\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: There is a message in the training data labeled with intent 'initial_registration'. This intent is not listed in your domain. You should need to add that intent to your domain file!\n",
      "  More info at https://rasa.com/docs/rasa/domain\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: The intent 'bot_challenge' is used in your stories, but it is not listed in the domain file. You should add it to your domain file!\n",
      "  More info at https://rasa.com/docs/rasa/domain\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: The intent 'goodbye' is used in your stories, but it is not listed in the domain file. You should add it to your domain file!\n",
      "  More info at https://rasa.com/docs/rasa/domain\n",
      "\u001b[0m2025-02-21 19:06:07 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating uniqueness of intents and stories...\n",
      "2025-02-21 19:06:07 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating utterances...\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: The action 'utter_goodbye' is used in the stories, but is not a valid utterance action. Please make sure the action is listed in your domain and there is a template defined with its name.\n",
      "  More info at https://rasa.com/docs/rasa/actions#utterance-actions\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: The action 'utter_iamabot' is used in the stories, but is not a valid utterance action. Please make sure the action is listed in your domain and there is a template defined with its name.\n",
      "  More info at https://rasa.com/docs/rasa/actions#utterance-actions\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: The utterance 'utter_initial_registration' is not used in any story or rule.\n",
      "\u001b[0m2025-02-21 19:06:07 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Story structure validation...\n",
      "2025-02-21 19:06:07 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.core.training.story_conflict\u001b[0m  - Considering all preceding turns for conflict analysis.\n",
      "2025-02-21 19:06:07 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - No story structure conflicts found.\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/.ipynb_checkpoints/rules-checkpoint.yml': \n",
      "Found intent 'goodbye' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/.ipynb_checkpoints/rules-checkpoint.yml': \n",
      "Found intent 'bot_challenge' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/rules.yml': \n",
      "Found intent 'goodbye' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/rules.yml': \n",
      "Found intent 'bot_challenge' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[94mThe configuration for pipeline and policies was chosen automatically. It was written into the config file at 'config.yml'.\u001b[0m\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'RegexFeaturizer'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'RegexFeaturizer'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'LexicalSyntacticFeaturizer'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'LexicalSyntacticFeaturizer'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.featurizers.sparse_featurizer.count_vectors_featurizer\u001b[0m  - 37 vocabulary items were created for text attribute.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.featurizers.sparse_featurizer.count_vectors_featurizer\u001b[0m  - 456 vocabulary items were created for text attribute.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'DIETClassifier'.\n",
      "2025-02-21 19:06:08 \u001b[1;30mERROR   \u001b[0m \u001b[34mrasa.nlu.classifiers.diet_classifier\u001b[0m  - \u001b[31mCannot train 'DIETClassifier'. Need at least 2 different intent classes. Skipping training of classifier.\u001b[0m\n",
      "2025-02-21 19:06:08 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'DIETClassifier'.\n",
      "2025-02-21 19:06:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'EntitySynonymMapper'.\n",
      "2025-02-21 19:06:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'EntitySynonymMapper'.\n",
      "2025-02-21 19:06:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'ResponseSelector'.\n",
      "2025-02-21 19:06:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.selectors.response_selector\u001b[0m  - Retrieval intent parameter was left to its default value. This response selector will be trained on training examples combining all retrieval intents.\n",
      "2025-02-21 19:06:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'ResponseSelector'.\n",
      "Processed rules: 100%|████████████| 4/4 [00:00<00:00, 1751.82it/s, # trackers=1]\n",
      "2025-02-21 19:06:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'MemoizationPolicy'.\n",
      "Processed trackers: 0it [00:00, ?it/s]\n",
      "2025-02-21 19:06:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'MemoizationPolicy'.\n",
      "2025-02-21 19:06:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'RulePolicy'.\n",
      "Processed trackers: 100%|███████████| 2/2 [00:00<00:00, 1021.26it/s, # action=5]\n",
      "Processed actions: 5it [00:00, 15791.81it/s, # examples=4]\n",
      "Processed trackers: 0it [00:00, ?it/s]\n",
      "Processed trackers:   0%|                                 | 0/2 [00:00<?, ?it/s]\n",
      "2025-02-21 19:06:09 \u001b[1;30mERROR   \u001b[0m \u001b[34mrasa.engine.graph\u001b[0m  - \u001b[31mError running graph component for node train_RulePolicy1.\u001b[0m\n",
      "\u001b[91mActionNotFoundException: Cannot access action 'utter_goodbye', as that name is not a registered action for this domain. Available actions are: \n",
      "\t - action_listen\n",
      "\t - action_restart\n",
      "\t - action_session_start\n",
      "\t - action_default_fallback\n",
      "\t - action_deactivate_loop\n",
      "\t - action_revert_fallback_events\n",
      "\t - action_default_ask_affirmation\n",
      "\t - action_default_ask_rephrase\n",
      "\t - action_two_stage_fallback\n",
      "\t - action_unlikely_intent\n",
      "\t - action_back\n",
      "\t - ...\n",
      "\t - action_extract_slots\n",
      "\t - utter_initial_registration\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!rasa train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bc5c9488-c108-4721-9f0e-f311a69a201d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/tracker_store.py:1044: MovedIn20Warning: \u001b[31mDeprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. \u001b[32mTo prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to \"sqlalchemy<2.0\". \u001b[36mSet environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message.\u001b[0m (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)\n",
      "  Base: DeclarativeMeta = declarative_base()\n",
      "usage: rasa [-h] [--version]\n",
      "            {init,run,shell,train,interactive,telemetry,test,visualize,data,export,x,evaluate}\n",
      "            ...\n",
      "rasa: error: unrecognized arguments: --skip-checkpoints\n"
     ]
    }
   ],
   "source": [
    "!rasa train --skip-checkpoints\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "42580ed8-948f-45a4-83a3-e4077f80e030",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: sqlalchemy<2.0 in /home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages (1.4.54)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in /home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages (from sqlalchemy<2.0) (3.1.1)\n"
     ]
    }
   ],
   "source": [
    "!pip install \"sqlalchemy<2.0\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "80aafe56-8cd9-4c9d-a850-f2fabed1c7fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/tracker_store.py:1044: MovedIn20Warning: \u001b[31mDeprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. \u001b[32mTo prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to \"sqlalchemy<2.0\". \u001b[36mSet environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message.\u001b[0m (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)\n",
      "  Base: DeclarativeMeta = declarative_base()\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/validation.py:134: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html\n",
      "  import pkg_resources\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/pkg_resources/__init__.py:3117: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.\n",
      "Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages\n",
      "  declare_namespace(pkg)\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/pkg_resources/__init__.py:3117: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('ruamel')`.\n",
      "Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages\n",
      "  declare_namespace(pkg)\n",
      "2025-02-21 19:19:39 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.cli.train\u001b[0m  - Started validating domain and training data...\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/tensorflow/lite/python/util.py:52: DeprecationWarning: jax.xla_computation is deprecated. Please use the AOT APIs.\n",
      "  from jax import xla_computation as _xla_computation\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Loading domain from 'domain.yml' failed. Using empty domain. Error: 'Failed to load domain specification from '/home/moringa/capstone_project_group2/domain.yml'. File not found!'\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/.ipynb_checkpoints/stories-checkpoint.yml': \n",
      "Found intent 'initial_registration' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/stories.yml': \n",
      "Found intent 'initial_registration' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "2025-02-21 19:19:42 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating intents...\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: There is a message in the training data labeled with intent 'initial_registration'. This intent is not listed in your domain. You should need to add that intent to your domain file!\n",
      "  More info at https://rasa.com/docs/rasa/domain\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: The intent 'initial_registration' is used in your stories, but it is not listed in the domain file. You should add it to your domain file!\n",
      "  More info at https://rasa.com/docs/rasa/domain\n",
      "\u001b[0m2025-02-21 19:19:42 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating uniqueness of intents and stories...\n",
      "2025-02-21 19:19:42 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating utterances...\n",
      "2025-02-21 19:19:42 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Story structure validation...\n",
      "Processed story blocks: 100%|█████| 2/2 [00:00<00:00, 1210.83it/s, # trackers=1]\n",
      "2025-02-21 19:19:42 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.core.training.story_conflict\u001b[0m  - Considering all preceding turns for conflict analysis.\n",
      "2025-02-21 19:19:42 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - No story structure conflicts found.\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/.ipynb_checkpoints/stories-checkpoint.yml': \n",
      "Found intent 'initial_registration' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[0m\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Issue found in 'data/stories.yml': \n",
      "Found intent 'initial_registration' in stories which is not part of the domain.\n",
      "  More info at https://rasa.com/docs/rasa/stories\n",
      "\u001b[94mThe configuration for pipeline and policies was chosen automatically. It was written into the config file at 'config.yml'.\u001b[0m\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Found a rule-based policy in your configuration but no rule-based training data. Please add rule-based stories to your training data or remove the rule-based policy (`RulePolicy`) from your your configuration.\n",
      "  More info at https://rasa.com/docs/rasa/rules\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'RegexFeaturizer'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'RegexFeaturizer'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'LexicalSyntacticFeaturizer'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'LexicalSyntacticFeaturizer'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.featurizers.sparse_featurizer.count_vectors_featurizer\u001b[0m  - 37 vocabulary items were created for text attribute.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.featurizers.sparse_featurizer.count_vectors_featurizer\u001b[0m  - 456 vocabulary items were created for text attribute.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'DIETClassifier'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mERROR   \u001b[0m \u001b[34mrasa.nlu.classifiers.diet_classifier\u001b[0m  - \u001b[31mCannot train 'DIETClassifier'. Need at least 2 different intent classes. Skipping training of classifier.\u001b[0m\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'DIETClassifier'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'EntitySynonymMapper'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'EntitySynonymMapper'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'ResponseSelector'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.selectors.response_selector\u001b[0m  - Retrieval intent parameter was left to its default value. This response selector will be trained on training examples combining all retrieval intents.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'ResponseSelector'.\n",
      "Processed story blocks: 100%|██████| 2/2 [00:00<00:00, 993.44it/s, # trackers=1]\n",
      "Processed story blocks: 100%|██████| 2/2 [00:00<00:00, 421.33it/s, # trackers=1]\n",
      "Processed story blocks: 100%|██████| 2/2 [00:00<00:00, 736.36it/s, # trackers=2]\n",
      "Processed story blocks: 100%|█████| 2/2 [00:00<00:00, 1170.29it/s, # trackers=3]\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'MemoizationPolicy'.\n",
      "Processed trackers: 100%|███████████| 1/1 [00:00<00:00, 1144.42it/s, # action=3]\n",
      "Processed actions: 3it [00:00, 4076.10it/s, # examples=3]\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'MemoizationPolicy'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'RulePolicy'.\n",
      "Processed trackers: 0it [00:00, ?it/s]\n",
      "Processed trackers: 100%|███████████| 1/1 [00:00<00:00, 2270.87it/s, # action=3]\n",
      "Processed trackers: 0it [00:00, ?it/s]\n",
      "Processed trackers: 100%|███████████████████████| 1/1 [00:00<00:00, 1860.00it/s]\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'RulePolicy'.\n",
      "2025-02-21 19:19:44 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'TEDPolicy'.\n",
      "Processed trackers: 100%|███████████| 4/4 [00:00<00:00, 2471.96it/s, # action=7]\n",
      "Epochs: 100%|█| 100/100 [00:17<00:00,  5.57it/s, t_loss=0.146, loss=0.0111, acc=\n",
      "2025-02-21 19:20:03 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'TEDPolicy'.\n",
      "2025-02-21 19:20:03 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'UnexpecTEDIntentPolicy'.\n",
      "2025-02-21 19:20:03 \u001b[1;30mWARNING \u001b[0m \u001b[34mrasa.shared.utils.common\u001b[0m  - \u001b[33mThe UnexpecTED Intent Policy is currently experimental and might change or be removed in the future 🔬 Please share your feedback on it in the forum (https://forum.rasa.com) to help us make this feature ready for production.\u001b[0m\n",
      "Processed trackers: 100%|███████████| 4/4 [00:00<00:00, 4782.56it/s, # intent=4]\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/engine/graph.py\", line 496, in __call__\n",
      "    output = self._fn(self._component, **run_kwargs)\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/policies/ted_policy.py\", line 722, in train\n",
      "    model_data, label_ids = self._prepare_for_training(\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/policies/ted_policy.py\", line 612, in _prepare_for_training\n",
      "    tracker_state_features, label_ids, entity_tags = self._featurize_for_training(\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/policies/policy.py\", line 241, in _featurize_for_training\n",
      "    state_features, label_ids, entity_tags = self.featurizer.featurize_trackers(\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/featurizers/tracker_featurizers.py\", line 389, in featurize_trackers\n",
      "    label_ids = self._convert_labels_to_ids(trackers_as_labels, domain)\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/featurizers/tracker_featurizers.py\", line 1007, in _convert_labels_to_ids\n",
      "    label_ids = [\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/featurizers/tracker_featurizers.py\", line 1008, in <listcomp>\n",
      "    [domain.intents.index(intent) for intent in tracker_intents]\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/featurizers/tracker_featurizers.py\", line 1008, in <listcomp>\n",
      "    [domain.intents.index(intent) for intent in tracker_intents]\n",
      "ValueError: 'initial_registration' is not in list\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/bin/rasa\", line 8, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/__main__.py\", line 133, in main\n",
      "    cmdline_arguments.func(cmdline_arguments)\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/cli/train.py\", line 61, in <lambda>\n",
      "    train_parser.set_defaults(func=lambda args: run_training(args, can_exit=True))\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/cli/train.py\", line 101, in run_training\n",
      "    training_result = train_all(\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/api.py\", line 105, in train\n",
      "    return train(\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/model_training.py\", line 207, in train\n",
      "    return _train_graph(\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/model_training.py\", line 286, in _train_graph\n",
      "    trainer.train(\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/engine/training/graph_trainer.py\", line 105, in train\n",
      "    graph_runner.run(inputs={PLACEHOLDER_IMPORTER: importer})\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/engine/runner/dask.py\", line 101, in run\n",
      "    dask_result = dask.get(run_graph, run_targets)\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/dask/local.py\", line 557, in get_sync\n",
      "    return get_async(\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/dask/local.py\", line 500, in get_async\n",
      "    for key, res_info, failed in queue_get(queue).result():\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/concurrent/futures/_base.py\", line 439, in result\n",
      "    return self.__get_result()\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/concurrent/futures/_base.py\", line 391, in __get_result\n",
      "    raise self._exception\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/dask/local.py\", line 542, in submit\n",
      "    fut.set_result(fn(*args, **kwargs))\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/dask/local.py\", line 238, in batch_execute_tasks\n",
      "    return [execute_task(*a) for a in it]\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/dask/local.py\", line 238, in <listcomp>\n",
      "    return [execute_task(*a) for a in it]\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/dask/local.py\", line 229, in execute_task\n",
      "    result = pack_exception(e, dumps)\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/dask/local.py\", line 224, in execute_task\n",
      "    result = _execute_task(task, data)\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/dask/core.py\", line 119, in _execute_task\n",
      "    return func(*(_execute_task(a, cache) for a in args))\n",
      "  File \"/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/engine/graph.py\", line 503, in __call__\n",
      "    raise GraphComponentException(\n",
      "rasa.engine.exceptions.GraphComponentException: Error running graph component for node train_UnexpecTEDIntentPolicy2.\n"
     ]
    }
   ],
   "source": [
    "!rasa train\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "91ac43ca-a3f6-4c0d-90e0-6804610fcc6c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "actions     credentials.yml  endpoints.yml\tmodels\t   rules.yml\n",
      "config.yml  data\t     group2_caps.ipynb\tREADME.md  tests\n"
     ]
    }
   ],
   "source": [
    "!ls /home/moringa/capstone_project_group2/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "caf17429-c107-4376-8587-dc47df7220b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "actions     credentials.yml  domain.yml     group2_caps.ipynb  README.md  tests\n",
      "config.yml  data\t     endpoints.yml  models\t       rules.yml\n"
     ]
    }
   ],
   "source": [
    "!ls /home/moringa/capstone_project_group2/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b685351-a267-4ca1-b12d-cf3acbfbfdae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/core/tracker_store.py:1044: MovedIn20Warning: \u001b[31mDeprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. \u001b[32mTo prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to \"sqlalchemy<2.0\". \u001b[36mSet environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message.\u001b[0m (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)\n",
      "  Base: DeclarativeMeta = declarative_base()\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/validation.py:134: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html\n",
      "  import pkg_resources\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/pkg_resources/__init__.py:3117: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.\n",
      "Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages\n",
      "  declare_namespace(pkg)\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/pkg_resources/__init__.py:3117: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('ruamel')`.\n",
      "Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages\n",
      "  declare_namespace(pkg)\n",
      "2025-02-21 19:25:06 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.cli.train\u001b[0m  - Started validating domain and training data...\n",
      "/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/tensorflow/lite/python/util.py:52: DeprecationWarning: jax.xla_computation is deprecated. Please use the AOT APIs.\n",
      "  from jax import xla_computation as _xla_computation\n",
      "2025-02-21 19:25:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating intents...\n",
      "2025-02-21 19:25:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating uniqueness of intents and stories...\n",
      "2025-02-21 19:25:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Validating utterances...\n",
      "2025-02-21 19:25:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - Story structure validation...\n",
      "Processed story blocks: 100%|█████| 2/2 [00:00<00:00, 1602.10it/s, # trackers=1]\n",
      "2025-02-21 19:25:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.core.training.story_conflict\u001b[0m  - Considering all preceding turns for conflict analysis.\n",
      "2025-02-21 19:25:09 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.validator\u001b[0m  - No story structure conflicts found.\n",
      "\u001b[94mThe configuration for pipeline and policies was chosen automatically. It was written into the config file at 'config.yml'.\u001b[0m\n",
      "\u001b[93m/home/moringa/anaconda3/envs/group2_capstone/lib/python3.9/site-packages/rasa/shared/utils/io.py:100: UserWarning: Found a rule-based policy in your configuration but no rule-based training data. Please add rule-based stories to your training data or remove the rule-based policy (`RulePolicy`) from your your configuration.\n",
      "  More info at https://rasa.com/docs/rasa/rules\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'RegexFeaturizer'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'RegexFeaturizer'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'LexicalSyntacticFeaturizer'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'LexicalSyntacticFeaturizer'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.featurizers.sparse_featurizer.count_vectors_featurizer\u001b[0m  - 37 vocabulary items were created for text attribute.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.featurizers.sparse_featurizer.count_vectors_featurizer\u001b[0m  - 456 vocabulary items were created for text attribute.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'CountVectorsFeaturizer'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'DIETClassifier'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mERROR   \u001b[0m \u001b[34mrasa.nlu.classifiers.diet_classifier\u001b[0m  - \u001b[31mCannot train 'DIETClassifier'. Need at least 2 different intent classes. Skipping training of classifier.\u001b[0m\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'DIETClassifier'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'EntitySynonymMapper'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'EntitySynonymMapper'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'ResponseSelector'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.nlu.selectors.response_selector\u001b[0m  - Retrieval intent parameter was left to its default value. This response selector will be trained on training examples combining all retrieval intents.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'ResponseSelector'.\n",
      "Processed story blocks: 100%|█████| 2/2 [00:00<00:00, 1589.96it/s, # trackers=1]\n",
      "Processed story blocks: 100%|█████| 2/2 [00:00<00:00, 1761.94it/s, # trackers=1]\n",
      "Processed story blocks: 100%|██████| 2/2 [00:00<00:00, 834.36it/s, # trackers=2]\n",
      "Processed story blocks: 100%|██████| 2/2 [00:00<00:00, 382.15it/s, # trackers=3]\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'MemoizationPolicy'.\n",
      "Processed trackers: 100%|███████████| 1/1 [00:00<00:00, 1915.21it/s, # action=3]\n",
      "Processed actions: 3it [00:00, 6400.26it/s, # examples=3]\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'MemoizationPolicy'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'RulePolicy'.\n",
      "Processed trackers: 0it [00:00, ?it/s]\n",
      "Processed trackers: 100%|███████████| 1/1 [00:00<00:00, 1221.76it/s, # action=3]\n",
      "Processed trackers: 0it [00:00, ?it/s]\n",
      "Processed trackers: 100%|████████████████████████| 1/1 [00:00<00:00, 815.85it/s]\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'RulePolicy'.\n",
      "2025-02-21 19:25:11 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'TEDPolicy'.\n",
      "Processed trackers: 100%|███████████| 4/4 [00:00<00:00, 1466.28it/s, # action=7]\n",
      "Epochs: 100%|█| 100/100 [00:18<00:00,  5.49it/s, t_loss=0.145, loss=0.00866, acc\n",
      "2025-02-21 19:25:30 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Finished training component 'TEDPolicy'.\n",
      "2025-02-21 19:25:30 \u001b[1;30mINFO    \u001b[0m \u001b[34mrasa.engine.training.hooks\u001b[0m  - Starting to train component 'UnexpecTEDIntentPolicy'.\n",
      "2025-02-21 19:25:30 \u001b[1;30mWARNING \u001b[0m \u001b[34mrasa.shared.utils.common\u001b[0m  - \u001b[33mThe UnexpecTED Intent Policy is currently experimental and might change or be removed in the future 🔬 Please share your feedback on it in the forum (https://forum.rasa.com) to help us make this feature ready for production.\u001b[0m\n",
      "Processed trackers: 100%|████████████| 4/4 [00:00<00:00, 909.53it/s, # intent=4]\n",
      "Epochs:  37%|▎| 37/100 [00:12<00:04, 15.18it/s, t_loss=0.121, loss=0.00121, acc="
     ]
    }
   ],
   "source": [
    "!rasa train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb21e2e2-e39e-4a4e-8ca1-dba61ac541b3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "group2_capstone",
   "language": "python",
   "name": "group2_capstone"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.21"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
